File: finger_exercises/chapter3/primes_to_1000.py

def is_prime(number):
    ans = True
    for i in range(2, number):
        if number % i == 0:
            ans = False
            return ans
    return ans
 
solution = 0        
for num in range(3, 1000, 2):
    if is_prime(num):
        solution += num

print(solution)

File: finger_exercises/chapter3/primes_to_1000.c
#include <stdio.h>

int main(void)
{
    int prime_sum = 0;
    int is_prime;

    for (int i = 3; i < 1000; i = i +2)
    {
        is_prime = 1;
        for (int j = 2; j < i; j++)
        {
            if (i % j == 0)
            {
                is_prime = 0;
            }
        }
        if (is_prime == 1)
        {
            prime_sum += i;
        }
    }
    printf("%d\n", prime_sum);
    return 0;
}